src/main/scala/IdNormalizer.scala

```diff
@@ -1,45 +1,57 @@
 package io.carrera.jsontoavroschema
 
+import Thing.JSchema
+
 import io.lemonlabs.uri._
 
 object IdNormalizer {
-  def normalizeIds(root: JsonSchema): Either[ResolutionError, JsonSchema] =
-    for {
-      baseUri <- root.id.toRight(ResolutionError("$id must be specified in root schema"))
-      resolved <- normalizeIds(root, baseUri)
-    } yield resolved
+  def normalizeIds(root: JSchema): Either[ResolutionError, JSchema] = {
+    root match {
+      case Left(bool) => Right(Left(bool))
```

Comment:
We're just propogating the `Left`, wrapped in a `Right`.
Is there a more elegant way to do this?

Reply:
It's fine. I don't like the alternatives I played with.